lib/assessment.js

```diff
@@ -8,30 +8,44 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-var min = function min() {
-
-};
+var min = reduce(function(previousValue, currentValue, index, array) {
+  return previousValue < currentValue ? previousValue : currentValue;
+)};
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
-var sum = function sum() {
-
+var sum = function sum(previous) {
+  return
 };
 
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
-var any = function any(array, callback, thisArg) {
+  var every = function every(array, callback, thisArg) {
+  var result = true;
+  array.forEach(function(currentValue, index, array) {
+
+    result = result && callback(currentValue, index, array);
+  });
 
+  return result;
```

Comment:
This is a good `every` function, but a bad `any` function. What `every` does is start out assuming something is true for all elements in an array, and tries to prove itself wrong. So `result` begins as `true` and the first `false` you AND with it makes it `false`. Nothing ANDed with `false` will ever yield `true`, so it stays `false` indefinitely afterwards.

For `any`, we want to use different logic. We want to start out assuming `false` and OR the return value of the callback with it as we apply it on each array element. Once we OR `true` with it, `result` now stores `true`, and anything ORed with `true` yields `true`. This allows us to measure that at least one element of the array causes the callback to return `true`.

3/4 credit